CMakeLists.txt

```diff
@@ -362,10 +374,15 @@ else()
     set(US_CXX_FLAGS "${US_CXX_FLAGS} -stdlib=libc++")
   endif()
 
+  # If not cross-compiling, turn on Stack Smashing Protection.
+  # This is disabled for ARM as it requires an additional dependecy on libssp.so.
+  if(NOT CMAKE_TOOLCHAIN_FILE)
```

Comment:
I think this should test for the target system name instead of testing for the presence of a toolchain file (which could be present without actually cross-compiling, I believe). Maybe somthing like
```
if (NOT CMAKE_SYSTEM_NAME STREQUAL "Android")
```
or
```
if (NOT CMAKE_SYSTEM MATCHES "arm")
```

Reply:
I agree. I think I may be able to make the check more specific and determine whether -Wstack-protector would work through a compile time check, instead of checking against a target system name. I think its conceivable that clients could have stack smashing available in their ARM toolchains.